templates/development-bucket.yaml

```diff
@@ -0,0 +1,101 @@
+AWSTemplateFormatVersion: 2010-09-09
+
+Description: >-
+  An S3 bucket for use in development. This bucket will serve as the landing
+  bucket for incoming data, mimicking the Bridge bucket.
+
+Parameters:
+
+  BucketName:
+    Type: String
+    Description: Name of the bucket.
+    Default: ''
+
+  ReadWriteAccessArns:
+    Type: List<String>
+    Default: ''
+    Description: >-
+      (Optional) List of IAM users and roles (strings) who
+      should have read- and write-access to the S3 bucket.
+
+  ReadOnlyAccessArns:
+    Type: List<String>
+    Default: ''
+    Description: >-
+      (Optional) List of IAM users and roles (strings)
+      who should only have read-access to the S3 bucket.
+
+Conditions:
+  HasReadWriteAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadWriteAccessArns], ""]]
+  HasReadOnlyAccessArns:
+    !Not [!Equals [!Join [",", !Ref ReadOnlyAccessArns], ""]]
+  HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
+
+Resources:
+
+  Bucket:
+    Type: AWS::S3::Bucket
+    DeletionPolicy: Delete
+    Properties:
+      BucketName: !If [HasBucketName, !Ref BucketName, !Ref 'AWS::NoValue']
+      AccessControl: Private
+      OwnershipControls:
+        Rules:
+          - ObjectOwnership: BucketOwnerPreferred
+
+  BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref Bucket
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Sid: AccountRead
+            Effect: 'Allow'
+            Principal:
+              AWS: !Sub '${AWS::AccountId}'
+            Action: 's3:GetObject'
+            Resource: !Sub 'arn:aws:s3:::${Bucket}/*'
+          - !If
+            - HasReadWriteAccessArns
+            - Sid: WriteAccess
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadWriteAccessArns
+              Action:
+                - 's3:PutObject*'
+              Condition:
+                StringEquals:
+                  s3:x-amz-acl: bucket-owner-full-control
+              Resource:
+                - !Sub 'arn:aws:s3:::${Bucket}/*'
+            - !Ref AWS::NoValue
+          - !If
+            - HasReadOnlyAccessArns
+            - Sid: ReadAccess
+              Effect: Allow
+              Principal:
+                AWS: !Ref ReadWriteAccessArns
+              Action:
+                - 's3:GetObject*'
+                - 's3:DeleteObject*'
+                - 's3:*MultipartUpload*'
+                - 's3:ListBucket*'
+                - 's3:GetBucketLocation'
+              Resource:
+                - !Sub 'arn:aws:s3:::${Bucket}'
+                - !Sub 'arn:aws:s3:::${Bucket}/*'
+            - !Ref AWS::NoValue
+
+Outputs:
+
+  BucketName:
+    Value: !Ref Bucket
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-BucketName'
+
+  BucketArn:
+    Value: !GetAtt Bucket.Arn
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-BucketArn'
```

Comment:
This is a _draft_ of the bucket we'll need. It may be complete, or not. We'll tweak this as we go -- it is a development bucket and does not need to be perfect now.